src/redux/Rockets/RocketsSlice.js

```diff
@@ -0,0 +1,55 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const fetchRockets = createAsyncThunk(
+  'rockets/fetchData',
+  async () => {
+    try {
+      const response = await axios.get('https://api.spacexdata.com/v3/rockets');
+      return response.data;
+    } catch (error) {
+      return error;
+    }
+  },
+);
+
+const initialState = {
+  isLoading: false,
+  error: false,
+  rockets: [],
+};
+
+const Rockets = createSlice({
+  name: 'Rockets',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(fetchRockets.pending, (state) => ({
+      ...state,
+      isLoagind: true,
+    }
+    ));
+    builder.addCase(fetchRockets.fulfilled, (state, action) => {
+      const rockets = [];
+      action.payload.forEach((rocket) => {
+        const newRocket = {
+          id: rocket.rocket_id,
+          name: rocket.rocket_name,
+          type: rocket.rocket_type,
+          image: rocket.flickr_images[0],
+          description: rocket.description,
+        };
+        rockets.push(newRocket);
+      });
+      state.rockets = rockets;
+    });
+    builder.addCase(fetchRockets.rejected, (state) => ({
+      ...state,
+      isLoagind: false,
```

Comment:
Be careful with that little typo, I guess you mean isLoading

Reply:
Solved! thank you for notice